### PR TITLE
docs: add validate warnings and refactor the account's functions page

### PR DESCRIPTION
**Pull request by ArielElp**
### Description of the Changes

Added warnings regarding important validations that should manually be added to `__execute__`.
Slightly refactored the page, and removed most of the last section since the function's signatures don't add new information.

### PR Preview URL

https://starknet-io.github.io/starknet-docs/pr-1302/architecture-and-concepts/accounts/account-functions/

### Check List

- [X] Changes have been done against main branch, and PR does not conflict
- [X] PR title follows the convention: `<docs/feat/fix/chore>(optional scope): <description>`, e.g: `fix: minor typos in code`

- - -
This change is [Reviewable](https://reviewable.io/reviews/starknet-io/starknet-docs/1302)

### CODE REVIEW

**Review by iliav-starkware — APPROVED** [review]
melekh

File: components/Starknet/modules/architecture-and-concepts/pages/accounts/account-functions.adoc [hunk 1]
```diff
@@ -3,53 +3,67 @@
...
-| `+__validate_deploy__+` a| Required when deploying an instance of an account contract with a `DEPLOY_ACCOUNT` _transaction_.
+| `+__execute__+` | Always required. The signatures of `+__validate__+` and `+__execute__+` must be identical.
+
+[WARNING]
+====
+At the moment of writing (Starknet 0.13.2), two critical validations must happen in `+__execute__+`, and their absence can lead to draining of the account's funds:
+
+(1) `assert!(get_caller_address().is_zero())`
+
+This asserts that the account's `execute` is not called from another contract, thus skipping validations (in later versions we may disallow calling `__execute__` from another contract at the protocol level)
```
Comment (odednaor) [inline components/Starknet/modules/architecture-and-concepts/pages/accounts/account-functions.adoc:23]:
```suggestion
This asserts that the account's `+__execute__+` is not called from another contract, thus skipping `+__validate__+` (in future Starknet versions, we may disallow calling `__execute__` from another contract at the protocol level)
```

File: components/Starknet/modules/architecture-and-concepts/pages/accounts/account-functions.adoc [hunk 2]
```diff
@@ -3,53 +3,67 @@
...
+
+(2) `assert!(get_tx_info().unbox().version.into() >= 1_u32)`
+
+This asserts that the transaction's version is at least 1, preventing your account from accepting v0 transactions. It's critical to explicitly disallow the deprecated v0 transaction type, as v0 transactions assume that the signature verification happens in `+__execute__+`, and are thus skipping `+__validate__+` entirely.
```
Comment (odednaor) [inline components/Starknet/modules/architecture-and-concepts/pages/accounts/account-functions.adoc:27]:
```suggestion
This asserts that the transaction's version is at least 1, preventing the account from accepting `INVOKE` v0 transactions. It is critical to explicitly disallow the deprecated v0 transaction type, as v0 transactions assume that the signature verification happens in `+__execute__+` and thus skip `+__validate__+` entirely.
```

**Review by odednaor — COMMENTED** [review]
@ArielElp Made some comments

**Review by ArielElp — COMMENTED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starknet-io/starknet-docs/1302)* status: all files reviewed, 2 unresolved discussions (waiting on @odednaor)

___
*[`components/Starknet/modules/architecture-and-concepts/pages/accounts/account-functions.adoc` line 23 at r2](https://reviewable.io/reviews/starknet-io/starknet-docs/1302#gh-1660102932:-O0wBPMd7W4latksp1_5:b-ds1lr7) ([raw file](https://github.com/starknet-io/starknet-docs/blob/e2d2ea49f0cea7da7cf817531b89b67ed62bd0cc/components/Starknet/modules/architecture-and-concepts/pages/accounts/account-functions.adoc#L23)):*
<details><summary><i>Previously, odednaor wrote…</i></summary><blockquote>

```suggestion
This asserts that the account's `+__execute__+` is not called from another contract, thus skipping `+__validate__+` (in future Starknet versions, we may disallow calling `__execute__` from another contract at the protocol level)
```
</blockquote></details>

replace the first, re second, I think it's easier if we say what's going wrong (no validations)
___
*[`components/Starknet/modules/architecture-and-concepts/pages/accounts/account-functions.adoc` line 27 at r2](https://reviewable.io/reviews/starknet-io/starknet-docs/1302#gh-1660103193:-O0wBx-xB5_gZfwPWx5i:b-896fix) ([raw file](https://github.com/starknet-io/starknet-docs/blob/e2d2ea49f0cea7da7cf817531b89b67ed62bd0cc/components/Starknet/modules/architecture-and-concepts/pages/accounts/account-functions.adoc#L27)):*
<details><summary><i>Previously, odednaor wrote…</i></summary><blockquote>

```suggestion
This asserts that the transaction's version is at least 1, preventing the account from accepting `INVOKE` v0 transactions. It is critical to explicitly disallow the deprecated v0 transaction type, as v0 transactions assume that the signature verification happens in `+__execute__+` and thus skip `+__validate__+` entirely.
```
</blockquote></details>

Done.

**Review by odednaor — APPROVED** [review]
lgtm

**Pull request merged**